src/sphinxcontrib/towncrier/_towncrier.py

```diff
@@ -25,6 +28,20 @@
 
 
 def get_towncrier_config(
+    project_path: Path,
+    final_config_path: Union[Path, None],
```

Comment:
Dropping support for older Pythons means we can use newer syntax.
```suggestion
    final_config_path: Path | None,
```